File: tools/parity/dispatch/terminal-quick-commands.ts
// TS dispatch for the terminal-quick-commands parity module. The shared TS bodies
// were DELETED (the Rust orca-agents core is the sole impl — napi in main, wasm in
// the renderer), so this adapter drives the SAME wasm through the single
// `terminalQuickCommandOp` boundary: the vectors' goldens pin that surface and the
// harness's TS-vs-Rust diff degenerates to wasm-vs-binary.
import { gitWasmOracle } from './orca-git-wasm-oracle'

export function dispatch(fn: string, input: unknown): unknown {
  return JSON.parse(gitWasmOracle().terminalQuickCommandOp(fn, JSON.stringify(input ?? null)))
}

File: rust/crates/orca-parity/src/modules/terminal_quick_commands.rs
//! Parity dispatch for the terminal quick-command helpers vs
//! `src/shared/terminal-quick-commands.ts`. Delegates to the shared JSON boundary
//! that the napi addon + wasm also run, so the oracle can never drift from them.

use serde_json::Value;

pub fn dispatch(function: &str, input: &Value) -> Value {
    orca_agents::terminal_quick_command_json::dispatch(function, input)
}
